articles/active-directory/develop/quickstart-v2-java-webapp.md

```diff
@@ -33,86 +32,124 @@ When you've completed the guide, your application will accept sign-ins of person
+> - `Enter_the_Client_Secret_Here` - is the **Client Secret** you created in **Certificates & Secrets** for the application
+you registered.
 
 #### Step 4: Run the code sample
-1. Run the code sample, and open a browser and navigate to *http://localhost:8080*.
-1. The front page contains a **sign-in** button. Click on the **sign-in** button to redirect to Azure Active Directory. The user will be prompted for their credentials.  
-1. After successfully authenticating on Azure Active Directory, they will be redirected to *http://localhost:8080/msal4jsamples/secure/aad*. They are officially signed in to the application, and the page should show information for the signed in account. It will also contain buttons for: 
+
+To run the project, you can either:
+
+- Run it directly from your IDE by using the embedded spring boot server
+- or package it to a WAR file using [maven](https://maven.apache.org/plugins/maven-war-plugin/usage.html) and deploy it to a J2EE container solution such as [Apache Tomcat](https://tomcat.apache.org/maven-plugin-trunk/tomcat6-maven-plugin/examples/deployment.html).
+
+##### Running from IDE
+
+1.If you are running the web application from an IDE, click on run, then navigate to the home page of the project. For this sample, the standard home page URL is <http://localhost:8080>
+
+1. The front page contains a **sign-in** button. Click on the **sign-in** button to redirect to Azure Active Directory. The
+user will be prompted for their credentials.  
+1. After successfully authenticating on Azure Active Directory, they will be redirected to
+*http://localhost:8080/msal4jsamples/secure/aad*. They are officially signed in to the application, and the page should show
+information for the signed in account. It will also contain buttons for:
     - *Sign Out*: Will sign out the current user from the application, and redirect them the home page.
-    - *Show Users*: Will acquire a token for the Microsoft Graph, then call the Microsoft Graph with the token attached to the request to get all of the users in the tenant.
+    - *Show User Info*: Will acquire a token for the Microsoft Graph, then call the Microsoft Graph with the token attached to
+    the request to get the basic information of the signed-in user.
 
 ## More information
 
 ### Getting MSAL
-MSAL4J is the library used to sign in users and request tokens used to access an API protected by the Microsoft identity Platform. 
-You can add MSAL4J to your application by using Maven or Gradle to manage your dependencies by making the following changes to the pom.xml or build.gradle file in your application. 
+
+MSAL4J is the library used to sign in users and request tokens used to access an API protected by the Microsoft identity
+Platform.
+You can add MSAL4J to your application by using Maven or Gradle to manage your dependencies by making the following changes to
+the pom.xml or build.gradle file in your application.
```

Comment:
How about: "Add MSAL4J to your application by using Maven or Gradle. Make the following changes your application's pom.xml (Maven) or build.gradle (Gradle) file.

Reply:
As it is a quickstart doc, yes I think this looks good.